### PR TITLE
Improve ContextMixin

### PR DESCRIPTION
**Pull request by ibrag8998**
I just added useful `get_context_updates` method in `ContextMixin`. Before this, if you want to update your context, you need to write:
```python
def get_context_data(self, *args, **kwargs):
    context = super().get_context_data(*args, **kwargs)
    context.update({"message": "egassem"})
    return context
```
And now, with this handy method, you can simple write the following:
```python
def get_context_updates(self):
    return {"message": "egassem"}
```

The key feature of writing this, while there is `extra_context` attribute, is that you can now access `self`. This means, you have access, for example, to `request`

### CODE REVIEW

File: django/views/generic/base.py [hunk 1]
```diff
@@ -27,8 +27,12 @@ def get_context_data(self, **kwargs):
         kwargs.setdefault('view', self)
         if self.extra_context is not None:
             kwargs.update(self.extra_context)
+        kwargs.update(self.get_context_updates())
         return kwargs
 
+    def get_context_updates(self):
```
Comment (auvipy) [inline django/views/generic/base.py:33]: what are the benefits of this? can you also provide tests to verify this changes?

Comment (ibrag8998) [reply django/views/generic/base.py:33]: Thanks for the response! This is not something that will change the way django works, it is just a little patch to save developers from writing unnecessary 2 lines of code every time they want to dynamically update context

Comment (auvipy) [reply django/views/generic/base.py:33]: it would be better to add test to demonstrate that

**Comment by felixxm** [comment]
@ibrag8998 Thanks for this patch, however I don't see much value in adding an extra hook. Overriding inherited methods with a `super()` call is a natural way in Python and it is also more explicit.

**Pull request closed without merging**